cmd/config_helper.go

```diff
@@ -107,20 +109,27 @@ func collectDependencies(pkg string, requires []string, providers map[string]str
 		depSet[provider] = true
 	}
 
-	deps := keys(depSet)
+	return sortedKeys(depSet), nil
+}
 
-	found := map[string]bool{pkg: true}
+func removeCyclicDependencies(allPackages []*bazeldnf.RPM) []*bazeldnf.RPM {
+	alreadyInstalled := make(map[string]bool)
```

Comment:
This should probably be nested inside of the outer loop?   The end product of this isn't going to remove cyclic dependency fan out, per se, it's going to remove instances of a given package from later packages in the list too?

Reply:
You mean it may lead to incomplete trees? if A depends on B and B depends on A, then Bazel will only know that A depends on B, so if you try to create an rpmtree for B you will have a non complete tree, to me that's an acceptable cost.

For instance Perl requires a few perl- <foo> packages and those also require perl. In this case I think it's acceptable that the maintainer of the BUILD file adds both when needed

Reply:
Ohh wait, I think I know what you mean, now that I realize there's a bug in my code. What I should do is for each dependency of A get the transitive dependency tree and if I find A again then break the cycle, which is not what my current code is doing, I'll rework it

Reply:
Right.  I think you just want to do a topological sort of the dependency fanout on per package basis.

Reply:
Cool, changed the process to take that into account

Reply:
After our Slack conversation it makes sense than only top level RPMs have their dependency tree listed, and the rest just voided